app/controllers/active_storage/blobs_controller.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+# Take a signed permanent reference for a blob and turn it into an expiring service URL for download.
+# Note: These URLs are publicly accessible. If you need to enforce access protection beyond the
+# security-through-obscurity factor of the signed blob references, you'll need to implement your own
+# authenticated redirection controller.
+class ActiveStorage::BlobsController < ActiveStorage::BaseController
+  include ActiveStorage::SetBlob
+  include Rails.application.routes.url_helpers
+
+  before_action :authenticate
+
+  def show
+    expires_in Rails.application.config.active_storage.service_urls_expire_in
+    redirect_to @blob.service_url(disposition: params[:disposition])
+  end
+
+  private
+
+  def authenticate
+    # All unauthenticated requests go back to home page
+    return redirect_to root_path, flash: {  alert: I18n.t('devise.failure.unauthenticated') } unless publisher_signed_in?
```

Comment:
I think the convention is `redirect_to... and return`

Reply:
Got it https://github.com/brave-intl/publishers/pull/1651/commits/1e66b8e48b42774f668e1461baa435224d4a88a2